lib/ansible/config/manager.py

```diff
@@ -606,3 +606,17 @@ def initialize_plugin_configuration_definitions(self, plugin_type, name, defs):
             self._plugins[plugin_type] = {}
 
         self._plugins[plugin_type][name] = defs
+
+    @staticmethod
+    def get_deprecated_msg_from_config(dep_docs, include_removal=False):
+
+        removal = ''
+        if include_removal:
+            if 'removed_at_date' in dep_docs:
+                removal = "Will be removed in a release after %(removed_at_date)s\n\t"
+            else:
+                removal = "Will be removed in: Ansible %(removed_in)s\n\t"
+
+        msg = "Reason: %(why)s\n\t" + removal + "Alternatives: %(alternatives)s"
```

Comment:
Changing `alternative` to `alternatives` in the format string won't be compatible with a lot of existing deprecations. If we do add + prefer a plural field, I feel like a list `alternatives` would look better either as a comma-separated list or without the inner quotes around items.

Reply:
alternatives was always the correct version, even the module validation had this already. I'm not sure why a bunch of 'singular' versions proliferated,j I would ad most add a 'deprecation on deprecation alternative' .

Reply:
Ah, okay. I do see singular all over the place.

Reply:
Docs to update:
* https://github.com/ansible/ansible-documentation/blob/devel/docs/docsite/rst/dev_guide/module_lifecycle.rst?plain=1#L35
* https://github.com/ansible/ansible-documentation/blob/devel/docs/docsite/rst/dev_guide/module_lifecycle.rst?plain=1#L66

Reply:
I don't see a deprecation, why mark as resolved? Breaking deprecations mid-deprecation seems bad.

Reply:
Example queries of things that are needlessly broken by this PR:
- https://github.com/search?q=repo%3Aansible-collections%2Fcommunity.general%20alternative%3A&type=code
- https://github.com/search?q=repo%3Aansible-collections%2Fansible.windows%20alternative%3A&type=code
- https://github.com/search?q=repo%3Aansible-collections%2Fcommunity.windows+alternative%3A&type=code

Before:
```
$ ansible-doc community.windows.win_domain_ou
[DEPRECATION WARNING]: community.windows.win_domain_ou has been deprecated. Use microsoft.ad.ou instead. This feature will be removed from community.windows in version 3.0.0. Deprecation warnings 
can be disabled by setting deprecation_warnings=False in ansible.cfg.
> COMMUNITY.WINDOWS.WIN_DOMAIN_OU    (/home/shertel/.ansible/collections/ansible_collections/community/windows/plugins/modules/win_domain_ou.py)

        Manage Active Directory Organizational Units Adds, Removes and Modifies Active Directory Organizational Units Task should be delegated to a Windows
        Active Directory Domain Controller

ADDED IN: version 1.8.0 of community.windows

DEPRECATED: 

        Reason: This module has been moved into the C(microsoft.ad) collection.
        Will be removed in: Ansible 3.0.0
        Alternatives: Use the M(microsoft.ad.ou) module instead.
... snip...
```

On this branch:
```
$ ansible-doc community.windows.win_domain_ou
[DEPRECATION WARNING]: community.windows.win_domain_ou has been deprecated. Use microsoft.ad.ou instead. This feature will be removed from community.windows in version 3.0.0. Deprecation warnings 
can be disabled by setting deprecation_warnings=False in ansible.cfg.
ERROR! Unable to retrieve documentation from 'community.windows.win_domain_ou'. Invalid deprecation documentation structure. 'alternatives'
```